### PR TITLE
REFACTOR-#2656: Update docs with new algebra operator names

### LINKED ISSUE #2656: Redesign Modin to align with DataFrame Algebra
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. What kind of performance improvements would you like to see with this new API?

Redesign Modin's architecture to implement the DataFrame Algebra proposed in the vision paper found [here](https://people.eecs.berkeley.edu/~totemtang/paper/Modin.pdf). Predecessor to #2204

### PR DESCRIPTION
**Pull request by RehanSD**
## What do these changes do?
Updates the documentation to reflect the changes in the naming of the algebra operators. The two changes reflected are:
`MapReduce` -> `TreeReduce`
`Reduction` -> `Reduce`

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/developer/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #2656 ? 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

**Review by devin-petersohn — COMMENTED** [review]
LGTM, please clear any conflicts

File: docs/flow/modin/core/dataframe/algebra.rst [hunk 1]
```diff
@@ -33,22 +33,22 @@ Uniformly apply a function argument to each partition in parallel.
 .. figure:: /img/map_evaluation.svg
     :align: center
 
-Reduction operator
+Reduce operator
 ------------------
```
Comment (YarShev) [inline docs/flow/modin/core/dataframe/algebra.rst:37]:
```suggestion
Reduce operator
---------------
```

File: docs/flow/modin/core/dataframe/algebra.rst [hunk 2]
```diff
@@ -33,22 +33,22 @@ Uniformly apply a function argument to each partition in parallel.
...
 concatenate partitions along the specified axis. Also, note that the execution engine expects
-that the reduction function returns a one dimensional frame.
+that the reduce function returns a one dimensional frame.
 
 .. figure:: /img/reduce_evaluation.svg
     :align: center
 
-MapReduce operator
+TreeReduce operator
 ------------------
```
Comment (YarShev) [inline docs/flow/modin/core/dataframe/algebra.rst:47]:
```suggestion
TreeReduce operator
-------------------
```

File: docs/flow/modin/core/dataframe/algebra.rst [hunk 3]
```diff
@@ -33,22 +33,22 @@ Uniformly apply a function argument to each partition in parallel.
...
 Applies an argument function that reduces specified axis into a scalar. First applies map function to each partition
-in parallel, then concatenates resulted partitions along the specified axis and applies reduction
+in parallel, then concatenates resulted partitions along the specified axis and applies reduce
 function. In contrast with `Map function` template, here you're allowed to change partition shape
-in the map phase. Note that the execution engine expects that the reduction function returns a one dimensional frame.
+in the map phase. Note that the execution engine expects that the reduce function returns a one dimensional frame.
 
 Binary operator
 ----------------
```
Comment (YarShev) [inline docs/flow/modin/core/dataframe/algebra.rst:54]:
```suggestion
Binary operator
---------------
```

**Review by YarShev — COMMENTED** [review]
Left minor suggestions. Otherwise, LGTM!

**Comment by RehanSD** [comment]
Thanks @YarShev! Changes applied!

**Review by devin-petersohn — APPROVED** [review]
LGTM

**Review by YarShev — APPROVED** [review]
@RehanSD, thank you, LGTM!

**Pull request merged**